### PR TITLE
nrf/os: Fix building with MICROPY_VFS enabled.

### PR DESCRIPTION
**Pull request by andrewleech**
Building the nrf/actinius_icarus board has recently been broken due to changes in the VFS configuration. I've seen this in my ci builds as such:
```
micropython/ports/nrf/modules/uos/moduos.c:90: undefined reference to `disk_ioctl'
ld: /tmp/firmware.elf.GCE5h4.ltrans1.ltrans.o:(.rodata+0xdc0): undefined reference to `mp_fat_vfs_type'
```
---

`#define MICROPY_VFS` is disabled by default for the nrf port, however this board has it enabled explicitly. 
@alextsam can you provide some more information about whether this was enabled intentionally?

The first commit here is the simplest one which #defines out fatfs dependencies from `ports/nrf/modules/uos/moduos.c` if `MICROPY_VFS_FAT` is not enabled.

Secondly, I've enabled `MICROPY_VFS_FAT` in mpconfigboard.h and `MICROPY_FATFS` in mpconfigboard.mk for actinius_icarus. @alextsam let me know if this was not desired on this board (I don't have one) and I can remove that commit again.

### CODE REVIEW

**Comment by dpgeorge** [comment]
@glennrub any comment on this?

**Comment by glennrub** [comment]
The `moduos.c` I have also addressed in #5472 / [moduos.c](https://github.com/micropython/micropython/pull/5472/commits/c84f5bd2ae31adb84b41bfc996df4b9f98dabd42)

I agree, as it stands today, without #5472 it would not build correctly, unless you set `MICROPY_VFS_FAT` during make invocation. And there is no documentation of this being required. But from what i know, offline, this was the intention.

Actinius also provides a sensor featherwing with SD card slot, that would benefit from having `FATFS` inbuilt. I would say that we enable the `FATFS` as you suggest to improve the support for the sensor board.

File: ports/nrf/boards/actinius_icarus/mpconfigboard.mk [hunk 1]
```diff
@@ -2,5 +2,6 @@ MCU_SERIES = m33
 MCU_VARIANT = nrf91
 MCU_SUB_VARIANT = nrf9160
 LD_FILES += boards/nrf9160_1M_256k.ld
+MICROPY_FATFS = 1
```
Comment (glennrub) [inline ports/nrf/boards/actinius_icarus/mpconfigboard.mk:5]: Should it be `MICROPY_VFS_FAT`, to allow `extmod/extmod.mk` to include the sources?

Comment (andrewleech) [reply ports/nrf/boards/actinius_icarus/mpconfigboard.mk:5]: Ah, that looks like the all-ports version of the same thing. `MICROPY_FATFS` includes the same files directly in micropython/ports/nrf/Makefile

Comment (andrewleech) [reply ports/nrf/boards/actinius_icarus/mpconfigboard.mk:5]: But yes, defining `MICROPY_FATFS = 1` in the mk file means it doesn't need to be defined in the header as well so does double duty and makes it more compatible with other ports.

**Comment by alextsam** [comment]
First of all I am sorry for the late response -I've been away on long holidays with very limited internet access.

I see that the issue is resolved in the meantime, but I just wanted to confirm that indeed the idea was to enable the use of the SD card on our extension board and indeed we were always invoking the build with `MICROPY_VFS_FAT` set. Thank you

**Comment by dpgeorge** [comment]
The build with MICROPY_VFS was fixed in 7a833edf37ed9714abdf5b7cf96d615fe877f3b9.  VfsLfs2 was enabled for this board in 55d4321c3eb1a99b7f135ca930878d0ce97620e6

**Pull request closed without merging**